Microsoft.O365.Security.Native.ETW/EventRecordMetadata.hpp

```diff
@@ -153,6 +153,68 @@ namespace Microsoft { namespace O365 { namespace Security { namespace ETW {
             return dest;
         }
 
+#pragma endregion
+
+#pragma region ExtendedData
+
+        /// <summary>
+        /// If the event's extended data contains an Argon container ID, retrieve it.
+        /// Can be expensive, avoid calling more than once per event.
+        /// </summary>
+        /// <returns>
+        /// A Guid representing the Argon container ID if the data is present. Null it's not present. 
+        /// Throws a ContainerIdFormatException if the container ID is present but parsing fails.
+        /// </returns>
+        virtual System::Nullable<System::Guid> GetContainerId()
+        {
+            // We are expecting format "00000000-0000-0000-0000-0000000000000", 32 hex digits with 4 hyphens
+            constexpr size_t CONTAINER_ID_DATA_LENGTH_IN_BYTES = 36;
+
+            size_t extended_data_count = record_->ExtendedDataCount;
```

Comment:
nit - use `auto` wherever possible

Reply:
Did you mean with respect to the type of extended_data_count?

Reply:
Yes, and in all other places where the compiler can infer the appropriate type.

Reply:
Thanks for the link. I've combed through the PR and updated with these guidelines in mind.